netbox_branching/merge_strategies/squash.py

```diff
@@ -0,0 +1,790 @@
+        return (
+            f"<CollapsedChange {self.model_class.__name__}:{obj_id}{suffix} "
+            f"action={self.final_action} changes={len(self.changes)}>"
+        )
+
+    def collapse(self, logger):
+        """
+        Collapse the list of ObjectChanges for this object into a single action.
+
+        A key point is that we only care about the final state of the object. Also
+        each ChangeObject needs to be correct so the final state is correct, i.e.
+        if we delete an object, there aren't going to be other objects still referencing it.
+
+        ChangeObject can have CREATE, UPDATE, and DELETE actions.
+        We need to collapse these changes into a single action.
+        We can have the following cases:
+           - CREATE + (any updates) + DELETE = skip entirely
+           - (anything other than CREATE) + DELETE = DELETE
+           - CREATE + UPDATEs = CREATE
+           - multiple UPDATEs = UPDATE
+        """
+        if not self.changes:
+            self.final_action = None
+            self.prechange_data = None
+            self.postchange_data = None
+            self.last_change = None
+            return
+
+        # Sort by time (oldest first)
+        self.changes = sorted(self.changes, key=lambda c: c.time)
+
+        has_delete = any(c.action == 'delete' for c in self.changes)
+        has_create = any(c.action == 'create' for c in self.changes)
+
+        logger.debug(f"  Collapsing {len(self.changes)} changes...")
+
+        if has_delete:
+            if has_create:
+                # CREATE + DELETE = skip entirely
+                logger.debug("  -> Action: SKIP (created and deleted in branch)")
+                self.final_action = 'skip'
+                self.prechange_data = None
+                self.postchange_data = None
+                self.last_change = self.changes[-1]
+                return
+            else:
+                # Just DELETE (ignore all other changes like updates)
+                # prechange_data: original state from first change
+                # postchange_data: postchange_data from DELETE ObjectChange
+                logger.debug(
+                    f"  -> Action: DELETE (keeping only DELETE, ignoring {len(self.changes) - 1} other changes)"
+                )
+                delete_change = next(c for c in self.changes if c.action == 'delete')
+                self.final_action = 'delete'
+                self.prechange_data = self.changes[0].prechange_data
+                self.postchange_data = delete_change.postchange_data  # Should be None for DELETE, but use actual value
+                self.last_change = delete_change
+                return
+
+        # No DELETE - handle CREATE or UPDATEs
+        first_action = self.changes[0].action
+        first_change = self.changes[0]
+        last_change = self.changes[-1]
+
+        # Created (with possible updates) -> single create
+        if first_action == 'create':
```

Comment:
I think we can just reference `has_create` here, as it's not possible to have any other action prior to a create. (This might be another good opportunity for a sanity check.)